test/e2e/BYOHDockerFile

```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE=ubuntu:20.04
+ARG BASE_IMAGE=ubuntu:22.04
```

Comment:
Is it absolutely neccesary?

Reply:
yep, it's must be the same as local where is binary was built. otherwise there is a problem with libc version
https://github.com/nebius/cluster-api-provider-bringyourownhost/blob/dea283a8de2fdc53ad5afc660a35c0ee6823e22c/test/e2e/BYOHDockerFile#L14